lib/plan-builder-generated.js

```diff
@@ -6942,6 +6990,154 @@ bindParam(...args) {
 
     }
 }
+/**
+  * PatchBuilder objects have methods.
+  * @namespace planBuilder.PatchBuilder
+  * @since 3.4.0
+  */
+class PlanPatchBuilder extends types.ServerType {
+
+  constructor(ns, fn, args) {
+      console.log("****** line 7001 ***********")
+      console.log(args)
+    super(ns, fn, args);
+  }
+
+/**
+    * Insert a new node after another node. Provides a client interface to a server function. See {@link http://docs.marklogic.com/op.insertAfter|op.insertAfter}
+    * @method planBuilder.PatchBuilder#insertAfter
+    * @since 3.4.0
+    * @param { XsString } [path] - The Patch Builder Plan. You can either use the XQuery =&gt; chaining operator or specify the variable that captures the return value from the previous operation.
+    * @param { Item } [node] - The path to insert the node after.
+    * @returns { planBuilder.PatchBuilder }
+    */
+insertAfter(...args) {
+    console.log("****** in insertAfter *******")
+    const namer = bldrbase.getNamer(args, 'path');
+    const paramdefs = [['path', [types.XsString], true, false], ['node', [types.Item, PlanColumn, PlanParam], false, true]];
+    const checkedArgs = (namer !== null) ?
+        bldrbase.makeNamedArgs(namer, 'PlanPatchBuilder.insertAfter', 2, new Set(['path', 'node']), paramdefs, args) :
+        bldrbase.makePositionalArgs('PlanPatchBuilder.insertAfter', 2, false, paramdefs, args);
+    return new PlanPatchBuilder('op', 'insert-after', checkedArgs);
+
+    }
+/**
+    * Insert a new node before another node. Provides a client interface to a server function. See {@link http://docs.marklogic.com/op.insertBefore|op.insertBefore}
+    * @method planBuilder.PatchBuilder#insertBefore
+    * @since 3.4.0
+    * @param { XsString } [path] - The Patch Builder Plan. You can either use the XQuery =&gt; chaining operator or specify the variable that captures the return value from the previous operation.
+    * @param { Item } [node] - The path to insert the node before.
+    * @returns { planBuilder.PatchBuilder }
+    */
+insertBefore(...args) {
+    const namer = bldrbase.getNamer(args, 'path');
+    const paramdefs = [['path', [types.XsString], true, false], ['node', [types.Item, PlanColumn, PlanParam], false, true]];
+    const checkedArgs = (namer !== null) ?
+        bldrbase.makeNamedArgs(namer, 'PlanPatchBuilder.insertBefore', 2, new Set(['path', 'node']), paramdefs, args) :
+        bldrbase.makePositionalArgs('PlanPatchBuilder.insertBefore', 2, false, paramdefs, args);
+    return new PlanPatchBuilder('op', 'insert-before', checkedArgs);
```

Comment:
I think the problem may be here - when a new PlanPatchBuilder is created, you need to retain all of the arguments on the current PlanPatchBuilder.

The Java Client does this via the private `addArg` method in `PatchBuilderImpl`. That builds a new list of args, using the existing list of args. It then calls the private constructor on lines 30-32 which creates a new "op/suboperators" object and adds all the args that are passed in.